lib/database.js

```diff
@@ -128,12 +135,39 @@ var MssqlDriver = function(options) {
             reject(err);
             return;
           }
-          resolve(db);
         });
+        resolve(db);
```

Comment:
Why are you resolving with the DB before we've even established if the connection was OK?  This was working before